tools/frr-reload.py

```diff
@@ -771,73 +764,41 @@ def load_contexts(self):
                     ctx_keys.append(line)
 
             elif (
-                line.startswith("vni ")
+                (line.startswith("vni ")
                 and len(ctx_keys) == 2
                 and ctx_keys[0].startswith("router bgp")
-                and ctx_keys[1] == "address-family l2vpn evpn"
-            ):
-
-                # Save old context first
-                self.save_contexts(ctx_keys, current_context_lines)
-                current_context_lines = []
-                sub_main_ctx_key = copy.deepcopy(ctx_keys)
-                log.debug(
-                    "LINE %-50s: entering sub-sub-context, append to ctx_keys", line
-                )
-                ctx_keys.append(line)
-
-            elif (
-                line.startswith("interface ")
+                and ctx_keys[1] == "address-family l2vpn evpn")
+                or
+                (line.startswith("interface ")
                 and len(ctx_keys) == 2
                 and ctx_keys[0].startswith("mpls ldp")
-                and ctx_keys[1].startswith("address-family")
+                and ctx_keys[1].startswith("address-family"))
             ):
 
                 # Save old context first
                 self.save_contexts(ctx_keys, current_context_lines)
                 current_context_lines = []
                 sub_main_ctx_key = copy.deepcopy(ctx_keys)
                 log.debug(
-                    "LINE %-50s: entering sub-sub-context, append to ctx_keys", line
+                    "LINE %-50s: entering sub-sub-context, append to ctx_keys",
+                    line
                 )
                 ctx_keys.append(line)
 
             elif (
-                line.startswith("traffic-eng")
+                (line.startswith("traffic-eng")
                 and len(ctx_keys) == 1
-                and ctx_keys[0].startswith("segment-routing")
-            ):
-
-                # Save old context first
-                self.save_contexts(ctx_keys, current_context_lines)
-                current_context_lines = []
-                log.debug(
-                    "LINE %-50s: entering segment routing sub-context, append to ctx_keys",
-                    line,
-                )
-                ctx_keys.append(line)
-
-            elif (
-                line.startswith("segment-list ")
+                and ctx_keys[0].startswith("segment-routing"))
+                or
+                (line.startswith("segment-list ")
                 and len(ctx_keys) == 2
                 and ctx_keys[0].startswith("segment-routing")
-                and ctx_keys[1].startswith("traffic-eng")
-            ):
-
-                # Save old context first
-                self.save_contexts(ctx_keys, current_context_lines)
```

Comment:
We don't do this in the "policy" case, can you really combine these two branches?

Reply:
I have combined the left, append the new commit. Please review, thanks.